### PR TITLE
Str\Grapheme\reverse() function

### PR DESCRIPTION
**Pull request by yivi**
Add Str\\Grapheme\reverse() to match Str\\reverse() and Str\Byte\reverse()

### CODE REVIEW

**Comment by yivi** [comment]
Regarding the coverage decrease... I'm afraid I do not yet know how to produce an input that `grapheme_strlen` would choke on.

Docs (https://www.php.net/manual/en/function.grapheme-strlen.php) say the function returns `int|false|null` with `false` and on failure (?) and, with no indication of when `null` would be returned.

I'm trying to see how to create a string that would be an invalid UTF-8 string to see if we can recover the lost coverage.

**Comment by yivi** [comment]
Ok, got it. Pushing recovered coverage. :)

File: tests/unit/Str/Grapheme/ReverseTest.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Tests\Unit\Str\Grapheme;
+
+use PHPUnit\Framework\TestCase;
+use Psl;
```
Comment (azjezz) [inline tests/unit/Str/Grapheme/ReverseTest.php]: we usually import the component, so `use Psl\Str\Byte;` + `use Psl\Str\Grapheme;` for consistency.

File: src/Psl/Str/Grapheme/reverse.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    $length = grapheme_strlen($string);
+
+    if ($length === false || $length === null) {
+        throw new InvalidArgumentException('Cannot get string length. Possibly invalid UTF-8 sequence');
+    }
+
+    while ($length-- > 0) {
+        $reversed .= grapheme_substr($string, $length, 1);
+    }
```
Comment (azjezz) [inline src/Psl/Str/Grapheme/reverse.php]: handling these errors shouldn't be done here, there's already wrappers for these grapheme_* functions.

```suggestion
    $length = length($string);

    while ($length-- > 0) {
        $reversed .= slice($string, $length, 1);
    }
```

**Review by azjezz — APPROVED** [review]

**Comment by azjezz** [comment]
Thank you @yivi 🎉

**Pull request merged**